app/javascript/components/namespace-form/index.jsx

```diff
@@ -0,0 +1,72 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { API } from '../../http_api';
+import MiqFormRenderer from '../../forms/data-driven-form';
+import schema from './namespace-form.schema';
+import miqRedirectBack from '../../helpers/miq-redirect-back';
+import handleFailure from '../../helpers/handle-failure';
+
+const getInitialValues = (isDomain, namespaceId, f) => {
+  miqSparkleOn();
+  API.get(`/api/automate_${isDomain ? 'domains' : 'namespaces'}/${namespaceId}?attributes=name,description,enabled,source`).then((result) => {
```

Comment:
you can destruct `result` = `({name, description ...)}` and then do `setInitialValues({name, description, enabled})`